### ISSUE TITLE
Support for Swift Package Manager (SPM)

### ISSUE THREAD

**Issue by heyzooi** [docusign/native-ios-sdk#119]
As a developer I would love to have support for Swift Package Manager (SPM) so I don't have to use CocoaPods or Carthage in order to use the SDK

**Comment by kollDocuSign** [comment]
Victor, 

My name is Koll with the DocuSign Developer Support team. I saw your post here but it sounds like the more appropriate area for this feature request would be on the DocuSign API Ideas site found here: https://feedback.docusign.com/forums/936834-apis

I would suggest posting your suggestion on the above site because it was created for the purpose of submitting Ideas that others might also be interested in.
Let me know if you have any questions or input on the above! 


Best, 
Koll

**Comment by heyzooi** [comment]
Just did... https://feedback.docusign.com/forums/936834-apis/suggestions/45157993--ios-sdk-support-for-swift-package-manager

**Comment by JMS2088** [comment]
Thank you so much

**Comment by bdrelling** [comment]
> Hello @heyzooi So we have SPM on our road map, hopefully will have it officially supported by end of year For now we have a workaround if you need to use our SDK into your app through SPM mentioned here [#105 (comment)](https://github.com/docusign/native-ios-sdk/issues/105#issuecomment-1027254850)
> 
> Let us know if you face any issues with this work around we can help getting you unblocked.

It's almost the end of the _following_ year, is there any more information on this? I'm currently evaluating the DocuSign SDK and lack of SPM support is a dealbreaker for us, so we're now trying to evaluate if it will be added anytime soon or if we should find an alternate solution.

There are three individual issues (#62, #105, and this one, #119) that are all closed, directing users to the DocuSign forums, but we (a) all already have GitHub accounts if we're here, (b) GitHub Issues is probably the first place all of us are checking about the status of SPM support, and (b) GitHub issue subscription notifies us about changes, so I think directing people over to the DocuSign forums feels a bit out-of-touch.

Please consider opening (or re-opening) a single issue for developers to track the status of SPM support, and ideally provide some context so that we understand if the DocuSign SDK is worth implementing or avoiding at this time.

Thanks.

**Comment by mmohareb** [comment]
Hello @bdrelling 
Thanks for your feedback, it's true we have multiple requests for this, but due to other prioritize this has not made the cut.
Another reason this has not been prioritized is the availability of multiple ways to integrate with the SDK including the possibility of creating a local Swift Package using the xcframework

Let us know if none of these solutions work for you and we can try to understand more about your use case and way of usage to unblock it

If you need to share details on a non public forum you can use docusignmobilesdk@docusign.com 

> for now you can create a local Swift package using binaryTarget and adding the DocuSignSDK.xcframework
> https://github.com/docusign/native-ios-sdk/blob/master/DocuSign.podspec#L32C26-L32C98
> 
> you can follow the documentation here
> https://developer.apple.com/documentation/swift_packages/distributing_binary_frameworks_as_swift_packages

**Comment by bdrelling** [comment]
The solution does not work as provided because SPM finds two XCFrameworks within a single zip and can only use one. This and a few other issues with Embedded Signing has steered us towards finding a better solution for our app that does not involve DocuSign.

I likely don't have more time to contribute to this, but if you just try it locally using your instructions on a version that contains both DocuSignSDK and DocuSignAPI, you can reproduce the error I saw.

Apple's documentation seems to imply only a single XCFramework can live inside a single zip when referenced.

Best of luck.

**Comment by mmohareb** [comment]
Sorry it didn't work for your case, but just to complete this thread, we were able to create a Swift package with both XCFrameworks and integrate it using SPM, I will attach a sample below.

In case you need more support with other issues you are facing with using the SDK, Please open a new ticket for the embedded signing issues you are facing.

![image](https://github.com/docusign/native-ios-sdk/assets/6322177/c5813d61-259a-45e2-904e-e802fe67b922)

**Comment by bdrelling** [comment]
@mmohareb yes, of course it would work by manually including the XCFrameworks in our repository. In my message above I said:

> The solution does not work as provided because SPM finds two XCFrameworks **within a single zip** and can only use one.

I am referring to your comment [here](https://github.com/docusign/native-ios-sdk/issues/119#issuecomment-1874602196) in which you link directly to the `.zip` listed in the `.podspec` and Apple's documentation which includes referencing a hosted, zipped archive containing a single XCFramework. My assumption, which I guess is incorrect, is that you were implying that users should be able to simply generate a checksum and reference the `.zip` directly -- which _should_ be possible and _would_ be infinitely easier, but only if you had two distinct `.zip` archives, each containing a single XCFRamework.

Instead, consumers are left to either (a) host it ourselves in the correct format, or (b) include the (bulky) DocuSign XCFrameworks in our own repo, neither of which is ideal. I would not have assumed that you were suggesting option (b) here, because `DocuSignSDK.xcframework` is 270mb and `DocuSignAPI.xcframework` is 290mb. The fact that you're suggesting consumers should casually add nearly half a gigabyte to their repositories is... concerning, to say the least.

I'll echo my earlier comments: 

A lack of _any_ SPM support in 2023 is an absolute no-go. Not including SPM support indicates to me (and likely most mobile developers) a lack of understanding of Swift/Apple, as even with the _gnarliest_ of closed-source frameworks, the work effort for getting a _minimum_ level of SPM support is absolutely trivial.

At its basic level, you could simply split the `.zip` up and include a `Package.swift` pointing at the hosted `.zip` archives _today_ and give your consumers SPM support.

The fact this sort of low-effort work is being deprioritized and pushed for _years_ seriously erodes my confidence in the quality and stability of the SDK as a whole. Given this and the poor UX of Embedded Signing, we won't be using the SDK going forward.

**Comment by mmohareb** [comment]
Thanks for the suggestions, will consider them when we prioritize the work for SPM.
